src/org/sosy_lab/java_smt/solvers/SolverLess/SolverLessFormulaCreator.java

```diff
@@ -0,0 +1,293 @@
+        new DummyType(DummyType.Type.INTEGER),
+        new DummyType(DummyType.Type.RATIONAL),
+        new DummyType(DummyType.Type.STRING),
+        new DummyType(DummyType.Type.REGEX));
+  }
+
+  @Override
+  public DummyType getBitvectorType(int bitwidth) {
+    return new DummyType(bitwidth);
+  }
+
+  @Override
+  public DummyType getFloatingPointType(FloatingPointType type) {
+    return new DummyType(type.getExponentSize(), type.getMantissaSize());
+  }
+
+  @Override
+  public DummyType getArrayType(DummyType indexType, DummyType elementType) {
+    return new DummyType(indexType.myType, elementType.myType);
+  }
+
+  @Override
+  public SMTLIB2Formula makeVariable(DummyType pDummyType, String varName) {
+    SMTLIB2Formula result = new SMTLIB2Formula(pDummyType);
+    result.setName(varName);
+    return result;
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <T extends Formula> FormulaType<T> getFormulaType(T formula) {
+    if (formula instanceof SMTLIB2Formula) {
+      SMTLIB2Formula solverLessFormula = (SMTLIB2Formula) formula;
+      switch (solverLessFormula.getFormulaType().myType) {
+        case BITVECTOR:
+          return (FormulaType<T>)
+              FormulaType.getBitvectorTypeWithSize(solverLessFormula.getBitvectorLength());
+        case FLOATING_POINT:
+          return (FormulaType<T>)
+              FormulaType.getFloatingPointType(
+                  solverLessFormula.getExponent(), solverLessFormula.getMantissa());
+        case ARRAY:
+          return (FormulaType<T>)
+              FormulaType.getArrayType(
+                  solverLessFormula.getFirstArrayParameter().getFormulaTypeForCreator(),
+                  solverLessFormula.getSecondArrayParameter().getFormulaTypeForCreator());
+        case RATIONAL:
+          return (FormulaType<T>) FormulaType.RationalType;
+        case STRING:
+          return (FormulaType<T>) FormulaType.StringType;
+        case REGEX:
+          return (FormulaType<T>) FormulaType.RegexType;
+        case INTEGER:
+          return (FormulaType<T>) FormulaType.IntegerType;
+        default:
+          return (FormulaType<T>) FormulaType.BooleanType;
+      }
+    }
+    if (formula instanceof BitvectorFormula) {
+      return (FormulaType<T>)
+          FormulaType.getBitvectorTypeWithSize(extractInfo(formula).getBitvectorLength());
+    }
+    if (formula instanceof ArrayFormula) {
+      return (FormulaType<T>)
+          FormulaType.getArrayType(
+              extractInfo(formula).getFirstArrayParameter().getFormulaTypeForCreator(),
+              extractInfo(formula).getSecondArrayParameter().getFormulaTypeForCreator());
+    }
+    return super.getFormulaType(formula);
+  }
+
+  @Override
+  public FormulaType<?> getFormulaType(SMTLIB2Formula formula) {
+    return formula.getFormulaTypeForCreator();
+  }
+
+  @Override
+  public <R> R visit(FormulaVisitor<R> visitor, Formula formula, SMTLIB2Formula f) {
+    return null;
```

Comment:
Never return `null`!
Also, you are missing a core implementation here for the functionality of the independent SMTLib2 layer.